snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -529,79 +530,26 @@ def prime(self, force=False) -> None:  # noqa: C901
-        #      mismatch checks.
-        linker_incompat = dict()  # type: Dict[str, str]
-        try:
-            linker = os.path.basename(
-                self._project_options.get_core_dynamic_linker(self._base))
-            for elf_file in elf_files:
-                if not elf_file.is_linker_compatible(linker=linker):
-                    linker_incompat[elf_file.path] = \
-                        elf_file.get_required_glibc()
-            if linker_incompat:
-                formatted_items = ['- {} (requires GLIBC {})'.format(k, v)
-                                   for k, v in linker_incompat.items()]
-                logger.warning(
-                    'The GLIBC version of the targeted core is {}. A newer '
-                    'libc will be required for the following files:'
-                    '\n{}'.format(linker, '\n'.join(formatted_items)))
-        except errors.SnapcraftMissingLinkerInBaseError as base_error:
-            if is_classic or classic_mangling_needed:
-                raise base_error
-
-        dynamic_linker = None
-        if linker_incompat or libc6_staged or classic_mangling_needed:
-            if not libc6_staged:
-                raise errors.StagePackageMissingError(package='libc6')
-            dynamic_linker = elf.find_linker(
-                root_path=self.primedir,
-                snap_base_path=self._snap_base_path)
-        elif is_classic:
-            dynamic_linker = self._project_options.get_core_dynamic_linker(
-                self._base, expand=False)
-
-        if dynamic_linker and not self._build_attributes.no_patchelf():
+        if self._build_attributes.no_patchelf():
             logger.warning(
-                'Files in this part are going to be patched to execute '
-                'correctly on diverse environments.\n'
-                'To disable this behavior set '
-                '`build-attributes: [no-patchelf]` for the part.')
-            elf_patcher = elf.Patcher(
-                dynamic_linker=dynamic_linker,
-                root_path=self.primedir,
-                preferred_patchelf_path=staged_patchelf_path)
-            files_to_patch = elf.get_elf_files_to_patch(elf_files)
-            for elf_file in files_to_patch:
-                elf_patcher.patch(elf_file=elf_file)
-        elif dynamic_linker and self._build_attributes.no_patchelf():
-            logger.warning(
-                'The following files are not going to be patched to work '
-                'correctly in the environment as '
+                'The primed files for part {!r} files are not going to be '
+                'verified for correctness nor patched to work correctly '
+                'in the environment if required as '
                 '`build-attributes: [no-patchelf]` is set for the '
-                'part:\n{}'.format(
-                    ''.join(['- {}\n'.format(e.path) for e in elf_files])))
+                'part.'.format(self.name))
```

Comment:
Yeah, I like this better. Although I think it could be worded a little shorter. Maybe:

"The primed files for part {!r} will not be verified for correctness or patched: `build-attributes: [no-patchelf]` is set."

Reasoning: removing a few redundant references to the part, and also assuming that folks know WHY we typically patch, so no need to make the message more verbose than necessary. Thoughts?

Reply:
agreed, how about the symmetric message for the other one?

Reply:
I'm not sure I understand, which other one? I assumed you were talking about [this one](https://github.com/snapcore/snapcraft/pull/2005/files#diff-660a35e421f136410b0d3e5168c7a62cL590) which you're getting rid of here.

Reply:
in 8b7c78582c68fa0f03a730af0c76db48e82e47b7
```
The primed files for part 'gotty' will not be verified for correctness or patched: build-attributes: [no-patchelf] is set.
```